src/main/java/org/vufind/solr/indexing/PrintBrowseHeadings.java

```diff
@@ -0,0 +1,255 @@
+package org.vufind.solr.indexing;
+
+//
+// Author: Mark Triggs <mark@teaspoon-consulting.com>
+//
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.charset.Charset;
+
+// Note that this version is coming from Solr!
+import org.apache.commons.codec.binary.Base64;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexNotFoundException;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.StoredFields;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.ConstantScoreQuery;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.TermQuery;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.search.TotalHitCountCollector;
+import org.apache.lucene.store.FSDirectory;
+import org.vufind.util.BrowseEntry;
+import org.vufind.util.Utils;
+
+
+public class PrintBrowseHeadings
+{
+    private SolrFieldIterator nonprefAuthFieldIterator;
+
+    IndexSearcher bibSearcher;
+    IndexSearcher authSearcher;
+
+    private String luceneField;
+
+    private String KEY_SEPARATOR = "\1";
+    private String RECORD_SEPARATOR = "\r\n";
+
+    /**
+     * Load headings from the index into a file.
+     *
+     * @param fieldIterator      Leech for pulling in headings
```

Comment:
Late comment: would be good to change this parameter description, since the Leech classes have all been renamed. Maybe something like:
"Iterator for pulling in headings from Solr matches"